sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -245,6 +246,57 @@ def test_make_blobs():
         assert_almost_equal((X[y == i] - ctr).std(), std, 1, "Unexpected std")
 
 
+def test_make_blobs_n_samples_list():
+    n_samples = [50, 30, 20]
+    X, y = make_blobs(n_samples=n_samples, n_features=2, random_state=0)
+
+    assert_equal(X.shape, (sum(n_samples), 2), "X shape mismatch")
+    assert_equal(y.shape, (sum(n_samples),), "y shape mismatch")
+    assert_equal(np.unique(y).shape, (3,), "Unexpected number of blobs")
+
+
+def test_make_blobs_n_samples_list_with_centers():
+    n_samples = [20, 20, 20]
+    centers = np.array([[0.0, 0.0], [1.0, 1.0], [0.0, 1.0]])
+    cluster_stds = np.array([0.05, 0.2, 0.4])
+    X, y = make_blobs(n_samples=n_samples, centers=centers,
+                      cluster_std=cluster_stds, random_state=0)
+
+    assert_equal(X.shape, (sum(n_samples), 2), "X shape mismatch")
+    assert_equal(y.shape, (sum(n_samples),), "y shape mismatch")
+    assert_equal(np.unique(y).shape, (3,), "Unexpected number of blobs")
+    for i, (ctr, std) in enumerate(zip(centers, cluster_stds)):
+        assert_almost_equal((X[y == i] - ctr).std(), std, 1, "Unexpected std")
+
+
+def test_make_blobs_error():
+    n_samples = [20, 20, 20]
+    centers = np.array([[0.0, 0.0], [1.0, 1.0], [0.0, 1.0]])
+    cluster_stds = np.array([0.05, 0.2, 0.4])
+    wrong_centers_msg = ("Length of `n_samples` not consistent "
+                         "with number of centers.")
+    assert_raise_message(ValueError, wrong_centers_msg,
+                         make_blobs, n_samples, centers=centers[:-1])
+    wrong_std_msg = ("Length of `clusters_std` not consistent with "
+                     "number of centers.")
+    assert_raise_message(ValueError, wrong_std_msg,
+                         make_blobs, n_samples,
+                         centers=centers, cluster_std=cluster_stds[:-1])
+    wrong_type_msg = "Parameter `centers` must be array-like."
+    assert_raise_message(ValueError, wrong_type_msg,
+                         make_blobs, n_samples, centers=3)
+
+
+def test_make_blobs_n_samples_list_centers_none():
```

Comment:
I would probably make a single test for n_samples being a list, tuple or int with parametrize

Reply:
@glemaitre currently the implementation does not take into account tuples. The input can only be an array or an int.
I am sorry, the name of the PR is a bit misleading. I fixed that!

Is it necessary to take tuples as an input?